cupyx/scipy/ndimage/_morphology.py

```diff
@@ -363,7 +363,8 @@ def binary_erosion(input, structure=None, iterations=1, mask=None, output=None,
 
     .. seealso:: :func:`scipy.ndimage.binary_erosion`
     """
-    structure, _, _ = _prep_structure(structure, input.ndim)
+    axes = _util._check_axes(axes, input.ndim)
+    structure, _, _ = _prep_structure(structure, len(axes))
```

Comment:
Couldn't there be fewer `axes` than `input.ndim`? If so, should we use `input.ndim` here?

```suggestion
    structure, _, _ = _prep_structure(structure, input.ndim)
```

Reply:
The intent is for the user-provided structure to have <= input.ndim dimensions (matching the subset of axes being filtered). CuPy is supposed to handle inserting any necessary singleton dimensions so that `ndim` axes are eventually present by the time the kernel is called.

Reply:
How will it know where to add the singleton dimensions without providing `input.ndim` and `axes`?

Reply:
`_prep_structure` just prepares the structure with a shape matching the number of filtered axes (which will be <= ndim). The actual code inserting the extra dimensions was already present in a `_utils.expand_footprint` utility function used in `_binary_erosion` here:
https://github.com/cupy/cupy/blob/418a9169bc9994574a22e6bfb07edb39d91899c6/cupyx/scipy/ndimage/_morphology.py#L189-L193

or for the grayscale morphology filters it is here:
https://github.com/cupy/cupy/blob/418a9169bc9994574a22e6bfb07edb39d91899c6/cupyx/scipy/ndimage/_filters.py#L933